src/sqlparser/src/parser.rs

```diff
@@ -1201,7 +1201,7 @@ impl Parser {
             Token::Mul | Token::Div | Token::Mod | Token::Concat => Ok(40),
             Token::DoubleColon => Ok(50),
             Token::ExclamationMark => Ok(50),
-            Token::LBracket => Ok(10),
+            Token::LBracket => Ok(21),
```

Comment:
This is not enough. `select 1*array[1][1]` will fail. I'm not sure but maybe using 50? (same as upstream's fix)

Reply:
Thank you very much for pointing out the upstream fix. I added the test as well @xxchan